cpp/cmake_modules/UseCython.cmake

```diff
@@ -184,4 +184,24 @@ function(cython_add_module _name pyx_target_name generated_files)
   add_dependencies(${_name} ${pyx_target_name})
 endfunction()
 
+execute_process(COMMAND ${PYTHON_EXECUTABLE} -m cython --version
```

Comment:
@kou Does this code for fetching the Cython version look ok?

Reply:
I took this snippet [from `scikit-build`](https://github.com/scikit-build/scikit-build/blob/main/skbuild/resources/cmake/FindCython.cmake#L61-L81).

Reply:
It'll work.

If we take a code to our project, we need to mention its license to `LICENSE.txt`.

I think that both of our `UseCython.cmake` and scikit-build's `FindCython.cmake` are based on DyND's one: https://github.com/libdynd/dynd-python/tree/master/cmake
We have a section for DyND in our `LICENSE.txt`:
https://github.com/apache/arrow/blob/e54ad41ee5e3689130cfd5cd7518a9e2ab4b1957/LICENSE.txt#L1845-L1884

Could you also add scikit-build's license information to `LICENSE.txt`?

BTW, can we use `check_cxx_compiler_flag()` like approach (try building with the flag to check availability) instead of version check approach? (Is it too complicated?)

Reply:
https://github.com/scikit-build/cython-cmake/ might be the more up to date source to reference

(something else, IMO we should also consider adopting that version entirely, but haven't check how much customizations we have added over time)

Reply:
`cython-cmake` has that exact same snippet: https://github.com/scikit-build/cython-cmake/blob/main/src/cython_cmake/cmake/FindCython.cmake#L66-L86.

Reply:
> can we use `check_cxx_compiler_flag()` like approach

Not sure how to do that. Try to compile a minimal `.pyx` with the flag and check `stdout`?

Reply:
Yes. I think that `try_compile()` https://cmake.org/cmake/help/latest/command/try_compile.html will help us if we use this approach.

Reply:
I had a look at this and tried to get it to work but failed. I think it's going to be significantly more CMake code to maintain than the simple version check. Unless I'm missing something on how to get `try_compile` to work with Cython, in which case I'll need some more help.

Reply:
OK. Let's avoid this approach. Thanks for trying it and sharing your experience.